### PR TITLE
Automatically generate a spree api key when a user receives the admin role.

### PR DESCRIPTION
**Pull request by athal7**
Given the recent change to only use api keys for admin api
authentication instead of the session cookie, an admin needs an api key
in order to do api things, and it does not receive one by default. It
seems kind to grant this to admin users when they become an admin, to
reduce the number of steps to give someone admin priveleges.

### CODE REVIEW

File: core/app/models/spree/role_user.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module Spree
+  class RoleUser < ActiveRecord::Base
```
Comment (jordan-brough) [inline core/app/models/spree/role_user.rb]: Nice. Would you mind also adding a migration with timestamp columns for this table, now that it's broken out as an activerecord model? @philbirt and I found that useful for the similar OrderPromotion change just recently.

**Comment by jordan-brough** [comment]
:+1:

File: core/app/models/spree/role_user.rb (hunk 1 above)
Comment (athal7) [inline core/app/models/spree/role_user.rb]: sure thing!

Comment (athal7) [inline core/app/models/spree/role_user.rb]: added

**Comment by athal7** [comment]
40b0efacde35e44555e1cbfb4da5296b8a3f03ab

**Pull request closed without merging**